### DISCUSSION TITLE
Set custom Header Content-Disposition for FileDownload

### DISCUSSION THREAD

**Discussion by masterElmar** [pocketbase/pocketbase#1717] [Q&A] [answered]
With the FileDownload [API Hooks](https://pocketbase.io/docs/event-hooks/#file-api-hooks) i can add and change the Header Tag for some definitions.
But when I tried to change the Content-Disposition with
```go
e.HttpContext.Response().Header().Set("Content-Disposition", "inline")
```
The API overrides the Header with the default configured in go.
I already read a [thread ](https://github.com/pocketbase/pocketbase/discussions/164) about that topic.
But I need this functionality to serve plain HTML with JS files for iframes.
With kind regards Elmar

**Comment by ganigeorgiev** [comment, marked as answer]
I'll consider making the headers "optional" in the next v0.13/v0.14 release by setting the defaults only if an explicit value is missing.

I'm not sure what is your use case, but in general I'd advice to not serve html/js inline using the file download endpoint, especially if the files are user uploaded and not vetted. If you want to serve static public assets, the better way would be to put them inside `pb_public` or to define a custom endpoint that will serve the files from a directory (eg. `e.Router.GET("/myfiles/*", apis.StaticDirectoryHandler(os.DirFS("/path/to/dir"), false))`).

If you really want to use the `OnFileDownloadRequest` hook, as a workaround for now you can write your own file serve handler, something like (_I haven't tested it_):

```go
package main

import (
	"log"
	"net/http"

	"github.com/pocketbase/pocketbase"
	"github.com/pocketbase/pocketbase/core"
	"github.com/pocketbase/pocketbase/tools/hook"
	"gocloud.dev/blob"
)

func main() {
	app := pocketbase.New()

	app.OnFileDownloadRequest().Add(func(e *core.FileDownloadEvent) error {
		fs, err := app.NewFilesystem()
		if err != nil {
			return err
		}
		defer fs.Close()

		file, err := fs.GetFile(e.ServedPath)
		if err != nil {
			return err
		}
		defer file.Close()

		// in the next release GetFile() will return directly the *blob.Reader and the type assertion will not be needed
		br := file.(*blob.Reader)

		res := e.HttpContext.Response()
		req := e.HttpContext.Request()

		// add whatever response headers you want...

		http.ServeContent(res, req, e.ServedName, br.ModTime(), br)

		return hook.StopPropagation
	})

	if err := app.Start(); err != nil {
		log.Fatal(err)
	}
}
```

**Reply by ganigeorgiev** [reply]
_Just to let you know that the headers can now be overwritten in the [`develop`](https://github.com/pocketbase/pocketbase/tree/develop) branch and will be included in v0.13 (there is no ETA). If you don't want to wait for v0.13 you can update your go.mod to temporary point to the [`develop`](https://github.com/pocketbase/pocketbase/tree/develop) branch (aka. `github.com/pocketbase/pocketbase@develop`)._
